src/vit-testing/vitup/src/mode/mock/rest/search.rs

```diff
@@ -73,74 +67,139 @@ async fn search_impl(
     };
     match table {
         Table::Challenges => {
```

Comment:
@cameron 
Can we consider refactoring the body of the `search_impl` function to several functions, making it modular? 
It has more than 150 lines which make really hard to understand what's going on there.